### PR TITLE
feat: support postgres geometry & BYTEA_ARRAY

### PR DESCRIPTION
**Pull request by xhwhis**

### CODE REVIEW

File: src/sql/arrow_sql_gen/postgres.rs [hunk 1]
```diff
@@ -721,6 +775,15 @@ fn map_column_type_to_data_type(column_type: &Type) -> Option<DataType> {
             DataType::Boolean,
             true,
         )))),
+        Type::BYTEA_ARRAY => Some(DataType::List(Arc::new(Field::new(
```
Comment (phillipleblanc) [inline src/sql/arrow_sql_gen/postgres.rs:778]: Would you be able to add a test for this so we can ensure it doesn't break in the future? Adding a new test method to https://github.com/datafusion-contrib/datafusion-table-providers/blob/main/tests/postgres/mod.rs and following the existing pattern for how `test_arrow_postgres_roundtrip` works should be fairly straightforward.

**Review by phillipleblanc — COMMENTED** [review]
Thanks for the contribution! Once this is updated with a test, I'll merge it.

File: src/sql/arrow_sql_gen/postgres.rs (hunk 1 above)
Comment (xhwhis) [reply src/sql/arrow_sql_gen/postgres.rs:778]: I have added tests.

**Review by phillipleblanc — APPROVED** [review]

**Comment by phillipleblanc** [comment]
@xhwhis Merged! Thanks again 🙏

**Pull request merged**